aten/src/ATen/cuda/CUDAGraph.cpp

```diff
@@ -115,6 +117,17 @@ void CUDAGraph::capture_begin(MempoolId_t pool/*=0*/) {
   // due to the capture status being updated _after_ a capture had already started.
   c10::cuda::CUDACachingAllocator::beginAllocateStreamToPool(capture_dev_, capture_stream_, mempool_id_);
 
+#ifdef USE_C10D_NCCL
+  // If the watchdog has remaining work enqueued, an event query on the remaining work will crash
+  // the graph capture.
+  while (!c10d::ProcessGroupNCCL::watchDogsDone()) {
```

Comment:
Should we have a way to disable this? I could see some people not happy with the sleeps here?

Reply:
This codepath is already only enabled in graph captures, and I suspect in more than half the cases `watchDogsDone` will return `true`---the busy wait is more guarding against a somewhat rare but unlucky case that can crash things.